packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -276,4 +278,36 @@ export const run = async () => {
   if (requiredEnvVars.RUN_STATE_BATCH_SUBMITTER === 'true') {
     loop(() => stateBatchSubmitter.submitNextBatch())
   }
+
+  // Initialize metrics server
+  const server = http.createServer(async (req, res) => {
+    req.on('error', (err) => {
+      logger.warn('Server encountered request error', {
+        err,
+      })
+      res.statusCode = 400
+      res.end('400: Bad Request')
+      return
+    })
+
+    res.on('error', (err) => {
+      logger.warn('Server encountered response error', {
+        err,
+      })
+    })
+
+    if (req.url === '/metrics') {
+      res.setHeader(
+        'Content-Type',
+        txBatchSubmitter.defaultMetrics.registry.contentType
+      )
+      res.end(await txBatchSubmitter.defaultMetrics.registry.metrics())
```

Comment:
@optimisticben @snario your feedback would be helpful here:

currently i'm only serving the transaction batch submitter's metrics since it's the one who care about most, but i'm wondering if the transaction batch submitter, state batch submitter, and batch submitter process should have separate `Metrics` instances at all. currently i'm initializing 3 of those classes in the file

https://github.com/ethereum-optimism/optimism/blob/ce7fa52b392f1932fc4f4ee10f4f22d23d9c0769/packages/batch-submitter/src/exec/run-batch-submitter.ts#L44-L47

https://github.com/ethereum-optimism/optimism/blob/ce7fa52b392f1932fc4f4ee10f4f22d23d9c0769/packages/batch-submitter/src/exec/run-batch-submitter.ts#L186-L189

https://github.com/ethereum-optimism/optimism/blob/ce7fa52b392f1932fc4f4ee10f4f22d23d9c0769/packages/batch-submitter/src/exec/run-batch-submitter.ts#L211-L214

is this best practice, or should they all share a registry?

Reply:
IMO Metrics registries should be tied to a _process_ and right now all three of these are in the same process.

Reply:
Agreed, sounds right to have a single metric registry.